weave/type_serializers/Audio/audio.py

```diff
@@ -0,0 +1,26 @@
+import wave
+
+from weave.trace import serializer
+from weave.trace.custom_objs import MemTraceFilesArtifact
+
+AUDIO_FILE_NAME = "audio.wav"
+
+
+def save(obj: wave.Wave_read, artifact: MemTraceFilesArtifact, name: str) -> None:
+    frames = obj.readframes(obj.getnframes())
+    params = obj.getparams()
+    with artifact.writeable_file_path(AUDIO_FILE_NAME) as fp:
+        wav_file = wave.open(fp, "wb")
+        wav_file.setparams(params)
+        wav_file.writeframes(frames)
+        wav_file.close()
+    obj.rewind()
+
+
+def load(artifact: MemTraceFilesArtifact, name: str) -> wave.Wave_read:
+    path = artifact.path(AUDIO_FILE_NAME)
+    return wave.open(path, "rb")
```

Comment:
you need to close otherwise it can cause resource leak

Reply:
Do you have a second to pair on this? I understand the concern and am not quite sure the correct way to remedy it.... @andrewtruong